perfkitbenchmarker/azure/azure_virtual_machine.py

```diff
@@ -187,18 +193,26 @@ def CreateScratchDisk(self, disk_spec):
     Args:
       disk_spec: virtual_machine.BaseDiskSpec object of the disk.
     """
-    if disk_spec.disk_type == disk.LOCAL:
-      self.local_disk_counter += disk_spec.num_striped_disks
-      if self.local_disk_counter > self.max_local_disks:
-        raise errors.Error('Not enough local disks.')
-
-    # Instantiate the disk(s) that we want to create.
-    disks = [azure_disk.AzureDisk(disk_spec, self.name)
-             for _ in range(disk_spec.num_striped_disks)]
+    disks = []
+
+    for _ in xrange(disk_spec.num_striped_disks):
+      data_disk = azure_disk.AzureDisk(disk_spec, self.name)
+      if disk_spec.disk_type == disk.LOCAL:
+        # Local disk numbers start at 1 (0 is the system disk).
+        data_disk.disk_number = self.local_disk_counter + 1
+        self.local_disk_counter += 1
+        if self.local_disk_counter > self.max_local_disks:
+          raise errors.Error('Not enough local disks.')
+      else:
+        # Remote disk numbers start at 1 + max_local disks (0 is the system disk
+        # and local disks occupy 1-max_local_disks).
```

Comment:
Confusion here w/ "1-max_local_disks" regarding arithmetic and range, similar to pull 347.